### ISSUE TITLE
directive on object ?

### ISSUE THREAD

**Issue by pandada8** [99designs/gqlgen#293]
### Expected Behaviour
run directive function on object level
### Actual Behavior
not run
### Minimal graphql.schema and models to reproduce
N/A

I want a filter-like directive which can remove some field based on the object and current logined user.
It's currently not possible because
1. I cannot access parent object when using FIELD directive
2. OBJECT-level directive is not supported

**Comment by vvakame** [comment]
Can you explain a sample schema?

**Comment by pandada8** [comment]
A simple schema would like this

```
@directive hideFieldForSomePeople on OBJECT

schema {
  query: Query
}

type Query {
  veryComplicatedQuery1: [Object!]
  veryComplicatedQuery2: [Object!]
  veryComplicatedQuery3: [Object!]
  veryComplicatedQuery4: Object
}

type Object @hideFieldForSomePeople {
  field: String!
  some_field_want_to_hide: String
}
```
but I think this can be fixed by #314.

**Comment by ibantoo** [comment]
i follow your above case, but it doesn't log the object of `type Object` , did you solve this @pandada8  ? need information about this
